pkg/controllers/maintenancelimit_controller.go

```diff
@@ -69,36 +69,19 @@ func (r *MaintenanceLimitReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
-func (r *MaintenanceLimitReconciler) getMaintenanceLimitsByType(ctx context.Context, ml *repairmanv1.MaintenanceLimit) (uint, uint, error) {
-	var limit uint
-	log := r.Log.WithValues("maintenancelimit", ml.Name)
-	if !strings.EqualFold(ml.Name, "node") {
-		return limit, limit, errors.New("unsupported maintenance type")
-	}
-
-	nodelist := &corev1.NodeList{}
-	err := r.List(ctx, nodelist)
-	if err != nil {
-		log.Error(err, "failed to list", "type", ml.Name)
-		return limit, limit, err
-	}
-
-	if len(nodelist.Items) <= 0 {
-		return limit, limit, nil
-	}
-
-	limit = (ml.Spec.Limit * uint(len(nodelist.Items))) / 100
-	limit = uint(math.Max(float64(limit), float64(1)))
-	return limit, uint(len(nodelist.Items)), nil
-}
-
 // UpdateMaintenanceLimits update maintenance limits if changed
 func (r *MaintenanceLimitReconciler) UpdateMaintenanceLimits(ctx context.Context, log logr.Logger, ml *repairmanv1.MaintenanceLimit) error {
-	limit, count, err := r.getMaintenanceLimitsByType(ctx, ml)
+	limit, err := r.getMaintenanceLimitsByType(ctx, log, ml)
 	if err != nil {
 		log.Error(err, "failed to get maintenance limits by type", "type", ml.Name)
 		return err
 	}
+	var nodeList *corev1.NodeList
+	err = r.getNodes(ctx, log, nodeList)
+	if err != nil {
+		log.Error(err, "failed to get maintenance limits by type", "type", ml.Name)
+	}
+	count := len(nodeList.Items)
```

Comment:
Do we want this to be the general count or the tighter count with policies applied?